src/main/java/com/openshift/internal/restclient/model/PersistentVolume.java

```diff
@@ -0,0 +1,276 @@
+	private static final String PV_CAPACITY = "spec.capacity.storage";
+	private static final String PV_RECLAIM_POLICY = "spec.persistentVolumeReclaimPolicy";
+	private static final String PV_NFS = "spec.nfs";
+	private static final String PV_HOST_PATH = "spec.hostPath";
+	private static final String PV_SPEC = "spec";
+	private static final String SERVER = "server";
+	private static final String PATH = "path";
+	private static final String READ_ONLY = "readOnly";
+
+	public PersistentVolume(ModelNode node, IClient client, Map<String, String[]> propertyKeys) {
+		super(node, client, propertyKeys);
+	}
+
+	/**
+	 * @param unit the designated unit. One of "Ki", "Mi", "Gi", "Ti", "Pi", "Ei".
+	 * @return 0, if conversion not possible. Otherwise the capacity in given units.
+	 * @see {@link PersistentVolume#convert(String, MemoryUnit)}
+	 * @throws IllegalArgumentException if not supported {@code unit} given.
+	 */
+	@Override
+	public long getCapacity(String unit) {
+		String capacity = asString(PV_CAPACITY);
+		return convert(capacity, MemoryUnit.valueOf(unit));
+	}
+
+	/**
+	 * @param unit the designated unit. One of {@link MemoryUnit}'s Ki, Mi, Gi, Ti, Pi, Ei.
+	 * @return 0, if conversion not possible. Otherwise the capacity in given units.
+	 * @see {@link PersistentVolume#convert(String, MemoryUnit)}
+	 * @throws IllegalArgumentException if not supported {@code unit} given.
+	 */
+	@Override
+	public long getCapacity(MemoryUnit unit) {
+		String capacity = asString(PV_CAPACITY);
+		return convert(capacity, unit);
+	}
+
+	/**
+	 * @return the capacity in bytes.
+	 */
+	@Override
+	public long getCapacity() {
+		String capacity = asString(PV_CAPACITY);
+		return Math.multiplyExact(convert(capacity, MemoryUnit.Ki), 1024L);
+	}
+
+	/**
+	 * Sets the capacity. There is no conversion between units.
+	 * @param capacity the capacity
+	 * @param unit the unit
+	 */
+	@Override
+	public void setCapacity(long capacity, MemoryUnit unit) {
+		get(PV_CAPACITY).set(capacity + unit.name());
+	}
+
+	/**
+	 * @return the unit in which the capacity is represent.
+	 *  @see {@link MemoryUnit}
+	 */
+	@Override
+	public MemoryUnit getCapacityUnit() {
+		return parseCapacityUnit(asString(PV_CAPACITY));
+	}
+
+	/**
+	 * @return access modes
+	 * @see {@link com.openshift.restclient.model.volume.PVCAccessModes}
+	 * @see
+	 */
+	@Override
+	public Set<String> getAccessModes() {
+		Set<String> modes = new HashSet<>();
+		ModelNode modelNode = get(PV_ACCESS_MODES);
+		if(!modelNode.isDefined() || !modelNode.getType().equals(ModelType.LIST)) return modes;
+		for (ModelNode node : modelNode.asList()) {
+			modes.add(node.asString());
+		}
+		return modes;
+	}
+
+	/**
+	 * Sets the access modes.
+	 * If there are any modes present, they are overridden by the {@code modes} parameter.
+	 * @param modes
+	 *             the access modes
+	 */
+	@Override
+	public void setAccessModes(String... modes) {
+		ModelNode accessModeNode = get(PV_ACCESS_MODES);
```

Comment:
Convenience setter for this...though it takes a 'Set' but could a new one could be added to take an array